lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_workspace_runtime.rb

```diff
@@ -136,6 +136,9 @@ def instantiate(uri, user, root = nil)
             save_current_state_info(@state_machine_objects.last) unless @state_machine_objects.empty?
             @state_machine_objects.push(obj.object_name)
             reset_state_info(obj.object_name)
+            is_state_machine = true
```

Comment:
@gmcculloug 
These are the 2 places where the is_state_machine is set to either true or false

Reply:
Right, but that's not my point.  The variable is only set in a context that may never be called.  We only get into that block if `scheme == "miqaedb"`.  If `scheme` is set to something else we never initialize the variable, yet it is still referenced in the `ensure` block.

Maybe a better way to point this out is to look at the `push` variable that is also referenced from the `ensure` block.  It is initialized above the `begin` statement on line 131.  I think the `is_state_machine` variable should be initialized to `false` there.